code/modules/projectiles/projectile/bullets.dm

```diff
@@ -275,6 +275,26 @@
 	icon = 'icons/obj/chemical.dmi'
 	icon_state = "syringeproj"
 
+/obj/item/projectile/bullet/dart/syringe/pierce_ignore
+	piercing = TRUE
+
+/obj/item/projectile/bullet/dart/syringe/pierce_ignore/on_hit(atom/target, blocked = 0, hit_zone)
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		if(blocked != INFINITY)
+			if(!(M.wear_suit.flags & PUNCTUREIMMUNE))
+				..()
+				reagents.trans_to(M, reagents.total_volume)
+				return TRUE
+			else
+				blocked = INFINITY
+				target.visible_message("<span class='danger'>[src] was deflected!</span>", \
+									"<span class='userdanger'>You were protected against [src]!</span>")
+	..(target, blocked, hit_zone)
+	reagents.set_reacting(TRUE)
+	reagents.handle_reactions()
+	return TRUE
```

Comment:
ok this is entirely copypasted from the parent except the pierce immune check. the pierce immune check should be covered in `can_inject` rather than here.